cpp_generation/templates/cpp/c_interface/c_adapters.j2

```diff
@@ -19,12 +19,10 @@ template<typename T, size_t Rank>
 auto read_data(T* data, int length, std::string_view name, nda::shape_of_rank<Rank> shape) {
   const auto size = shape.flat_max() + 1;
   if (length != size) {
-    throw std::invalid_argument(
-      std::format(
-        "Input data '{}' is the wrong size. Received array of length '{}', expected '{}'.",
-        name, length, size
-      )
-    );
+    std::ostringstream oss;
+    oss << "Input data '" << name << "' is the wrong size. "
```

Comment:
I'm sad to see the `format` go not as tidy https://youtu.be/AvqzGaZ-GoM

But I guess makes the building tidier :D